### PR TITLE
Long running dump and restore capability

### LINKED ISSUE #6527: API cleanup: dump/restoreRef as long running operation
Dump/resource ref API are currently blocking API, we should enable posting the request and query for status to support long running dump or restore operations.

### PR DESCRIPTION
**Pull request by nopcoder**
**New dump and restore refs API**

A new API for dumping and restoring refs. The API allows users to submit requests and then pull the status of the task and the complete information about the request once it is finished.

The new API is used by the `lakectl refs-dump` and `lakectl refs-restore` commands.

The new API offers a long running operations without the risk of timeout dropping the request.

Close #6527

### CODE REVIEW

**Review by johnnyaug — APPROVED** [review]
Awesome. Not blocking, but please see comments.

File: pkg/catalog/catalog.proto [hunk 1]
```diff
@@ -22,3 +22,38 @@ message Entry {
 	AddressType address_type = 6;
 	string content_type = 7;
 }
+
+// Task is a generic task status message
+message Task {
+	string id = 1;
+	bool completed = 2;
```
[the remark this answers is not in the document]
Comment (ozkatz) [reply pkg/catalog/catalog.proto:29]: `bool running = 2;` ?

File: api/swagger.yml [hunk 2]
```diff
@@ -2686,6 +2733,125 @@ paths:
...
+          $ref: "#/components/responses/Unauthorized"
+        404:
+          $ref: "#/components/responses/NotFound"
+        default:
+          $ref: "#/components/responses/ServerError"
+    get:
+      tags:
+        - repositories
+      operationId: restoreStatus
+      summary: Restore repository (tags, commits, branches) from object store
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply api/swagger.yml:2830]: Thanks

[the remark this answers is not in the document]
Comment (johnnyaug) [reply api/swagger.yml:2830]: Not done?

File: pkg/api/controller_test.go [hunk 3]
```diff
@@ -4527,3 +4527,145 @@ func TestController_GarbageCollectionRules(t *testing.T) {
...
+		if response.JSON404 == nil {
+			t.Fatalf("Expected 404 (not found) response, got %s", response.Status())
+		}
+	})
+}
+
+// pullRestoreStatus polls the restore status endpoint until the restore is complete or times out.
+// test will fail in case of error.
+// will return nil in case of timeout.
+func pullRestoreStatus(t *testing.T, clt apigen.ClientWithResponsesInterface, repo string, taskID string) *apigen.RepositoryRestoreStatus {
```
[the remark this answers is not in the document]
Comment (johnnyaug) [reply pkg/api/controller_test.go:4662]: There are still occurrences of "pull" instead of "poll".

**Review by johnnyaug — APPROVED** [review]

File: api/swagger.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nopcoder) [reply api/swagger.yml:2830]: think I updated and override the doc version

**Pull request merged**